### PR TITLE
1. fix the multi results when use 'ioredis' module; 2. catch the erro…

### PR DESCRIPTION
**Pull request by dearpeach**
…r when use JSON.parse. because when the data is string, JSON.parse will throw error

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -80,7 +80,12 @@ function buildCache (client, opts) {
           return asPromise(client.zrem.bind(client), ZSET_KEY, key)
             .then(() => null);
         }
-        return JSON.parse(results[0]);
+
+        if (Array.isArray(results[0])) {
```
Comment (facundoolano) [inline index.js:84]: looks like you are doing this check twice, can you extract it to a function?

File: index.js [hunk 2]
```diff
@@ -195,7 +200,24 @@ function buildCache (client, opts) {
...
+        if (Array.isArray(res)) {
+          return JsonParse(res[1]);
+        } else {
+          return JsonParse(res);
+        }
+      });
+    });
+
+  const JsonParse = (str) => {
+    try {
```
Comment (facundoolano) [inline index.js:214]: console.error is probably not the way to go for a library. Why wouldn't you throw an error if the parsing fails?

**Comment by facundoolano** [comment]
Can you add a commented out line like [this one](https://github.com/facundoolano/redis-lru/blob/master/test/index.js#L5) for ioredis, and make sure tests pass if you use it with a running redis instance?

**Review by facundoolano — CHANGES_REQUESTED** [review]

**Comment by emeraldarcher** [comment]
What's the status on this - I'm running into the same problem and would like to get this PR in.

**Comment by facundoolano** [comment]
@emeraldarcher  I left some change requests that weren't addressed. Feel free to add them yourself to this PR and I'll review again.

**Comment by emeraldarcher** [comment]
> @emeraldarcher I left some change requests that weren't addressed. Feel free to add them yourself to this PR and I'll review again.

How would I go about doing that when I don't have access to either of the repos?

**Comment by facundoolano** [comment]
Fork and pull request

File: index.js (hunk 1 above)
Comment (emeraldarcher) [reply index.js:84]: > Can you add a commented out line like [this one](https://github.com/facundoolano/redis-lru/blob/master/test/index.js#L5) for ioredis, and make sure tests pass if you use it with a running redis instance?

I tried doing this with the code as is and several tests fail. I'm finding no discernable difference but tests seem to work if I make the max size of the LRU one bigger then the tests originally had. For some reason, the first item (I'm looking at the get or set tests max size test) `k1` is always null unless I bump the size of the LRU up from 3 to 4. Any ideas?

**Comment by facundoolano** [comment]
closing in favor of #14

**Pull request closed without merging**